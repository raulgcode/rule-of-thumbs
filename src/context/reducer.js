export const LOAD_USERS = "LOAD_USERS";
export const DO_VOTE = "DO_VOTE";
export const RESET_VOTE = "RESET_VOTE";
export const CHANGE_VIEW = "CHANGE_VIEW";

export const initialState = {
  users: [],
  votedUsers: new Set(),
  view: "grid",
};

export const appReducer = (state, action) => {
  switch (action.type) {
    case LOAD_USERS:
      return { ...state, users: action.payload.users };
    case DO_VOTE: {
      const user = state.users.find(({ id }) => id === action.payload.user.id);
      user.votes = action.payload.user.votes;
      state.votedUsers.add(user.id);
      return { ...state };
    }
    case RESET_VOTE: {
      state.votedUsers.delete(action.payload.user.id);
      return { ...state };
    }
    case CHANGE_VIEW:
      return { ...state, view: action.payload.view };
    default:
      return state;
  }
};
