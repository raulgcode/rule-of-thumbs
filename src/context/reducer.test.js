import { appReducer } from "./reducer";

describe("reducer", () => {
  it("should return default state object", () => {
    const ANY_ACTION_TYPE = "ANY_ACTION_TYPE";
    const defaultState = {
      users: [],
      votedUsers: new Set(),
      view: "grid",
    };
    const state = appReducer(defaultState, { type: ANY_ACTION_TYPE });
    expect(state).toMatchSnapshot();
  });
});
