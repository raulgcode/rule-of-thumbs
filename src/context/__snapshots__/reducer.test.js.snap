// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducer should return default state object 1`] = `
Object {
  "users": Array [],
  "view": "grid",
  "votedUsers": Set {},
}
`;
